core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -649,35 +578,22 @@ open class TileGroup(
     }
 
     private fun updateTileColor(isViewable: Boolean) {
-        var color =
-                if (tileSetStrings.tileSetConfig.useColorAsBaseTerrain)
-                    tileInfo.getBaseTerrain().getColor()
-                else Color.WHITE.cpy() // no need to color it, it's already colored
-
-        if (!isViewable) color = color.lerp(tileSetStrings.tileSetConfig.fogOfWarColor, 0.6f)
-        for(image in tileBaseImages) image.color = color
-    }
+        val baseTerrainColor = when {
+            tileSetStrings.tileSetConfig.useColorAsBaseTerrain && !isViewable -> tileInfo.getBaseTerrain().getColor().lerp(tileSetStrings.tileSetConfig.fogOfWarColor, 0.6f)
+            tileSetStrings.tileSetConfig.useColorAsBaseTerrain -> tileInfo.getBaseTerrain().getColor()
+            !isViewable -> Color.WHITE.cpy().lerp(tileSetStrings.tileSetConfig.fogOfWarColor, 0.6f)
+            else -> Color.WHITE.cpy()
+        }
 
-    private fun updateTerrainFeatureImage() {
-        if (tileInfo.terrainFeatures != terrainFeatures) {
-            terrainFeatures.clear()
-            terrainFeatures.addAll(tileInfo.terrainFeatures)
-            if (terrainFeatureOverlayImage != null) terrainFeatureOverlayImage!!.remove()
-            terrainFeatureOverlayImage = null
-
-            for (terrainFeature in terrainFeatures) {
-                val terrainFeatureOverlayLocation = tileSetStrings.orFallback { getTerrainFeatureOverlay(terrainFeature) }
-                if (!ImageGetter.imageExists(terrainFeatureOverlayLocation)) return
-                terrainFeatureOverlayImage = ImageGetter.getImage(terrainFeatureOverlayLocation)
-                terrainFeatureLayerGroup.addActor(terrainFeatureOverlayImage)
-                terrainFeatureOverlayImage!!.run {
-                    setSize(30f, 30f)
-                    setColor(1f, 1f, 1f, 0.5f)
-                    center(this@TileGroup)
-                }
-            }
+        val color = when {
+            !isViewable -> Color.WHITE.cpy().lerp(tileSetStrings.tileSetConfig.fogOfWarColor, 0.6f)
+            else -> Color.WHITE.cpy()
```

Comment:
I'm not entirely sure what happens here, doesn't this result in a change in behaviour, where all tiles not currently shown tiles are grey instead of using a greyed out version the base terrain colour?

Reply:
I did change how "useColorAsBaseTerrain" works. It now only colors a base hexagon instead of everything on a tile. That's why I've split the local color variable in the updateTileColor function into "baseTerrainColor" and "color". "baseTerrainColor" is applied only to the first tileBaseImage which if "useColorAsBaseTerrain" is true is a default Hexagon.